src/OpenTK.Platform.Native/X11/X11AbstractionLayer.Window.cs

```diff
@@ -138,21 +229,66 @@ public void Destroy(WindowHandle handle)
         public string GetTitle(WindowHandle handle)
         {
             var window = handle.As<XWindowHandle>(this);
-            string str = string.Empty;
-            unsafe
+            string str;
+            IntPtr name;
+
+            if (IsWindowManagerFreedesktop)
+            {
+                XGetWindowProperty(
+                    window.Display,
+                    window.Window,
+                    _atoms![KnownAtoms._NET_WM_NAME],
+                    0,
+                    long.MaxValue,
+                    false,
+                    _atoms![KnownAtoms.UTF8_STRING],
+                    out _,
+                    out _,
+                    out long count,
+                    out _,
+                    out name);
+                str = Marshal.PtrToStringUTF8(name) ?? string.Empty;
+                XFree(name);
+            }
+            else
             {
-                XFetchName(window.Display, window.Window, out byte* name);
-                str = Marshal.PtrToStringAnsi((IntPtr)name) ?? string.Empty;
-                XFree((IntPtr)name);
+                XFetchName(window.Display, window.Window, out name);
             }
 
+            str = Marshal.PtrToStringAnsi(name) ?? string.Empty;
+            XFree(name);
+
             return str;
         }
 
         public void SetTitle(WindowHandle handle, string title)
         {
             var window = handle.As<XWindowHandle>(this);
-            XStoreName(window.Display, window.Window, title);
+            if (IsWindowManagerFreedesktop)
```

Comment:
We shouldn't need to check this. We should just set both properties. Will reduce the amount of code we have to write as well as the complexity. We should do this for all "set" functions.